flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/AlterTableCompactITCase.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.connector;
+
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.store.file.utils.BlockingIterator;
+import org.apache.flink.table.store.file.utils.SnapshotFinder;
+import org.apache.flink.types.Row;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.apache.flink.table.store.file.FileStoreOptions.relativeTablePath;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** ITCase for 'ALTER TABLE ... COMPACT'. */
+public class AlterTableCompactITCase extends FileStoreTableITCase {
+
+    @Override
+    protected List<String> ddl() {
+        return Arrays.asList(
+                "CREATE TABLE IF NOT EXISTS T0 (f0 INT, f1 STRING, f2 DOUBLE)",
+                "CREATE TABLE IF NOT EXISTS T1 ("
+                        + "f0 INT, f1 STRING, f2 STRING) PARTITIONED BY (f1)",
+                "CREATE TABLE IF NOT EXISTS T2 ("
+                        + "f0 INT, f1 STRING, f2 STRING) PARTITIONED BY (f1, f0)");
+    }
+
+    @Test
+    public void testNonPartitioned() throws Exception {
+        bEnv.executeSql("INSERT INTO T0 VALUES (1, 'Pride and Prejudice', 9.0), (2, 'Emma', 8.5)")
```

Comment:
Nit: we can use `sql` to replace the `bEnv.executeSql` once we got this PR merged: https://github.com/apache/flink-table-store/pull/126/files#diff-9446f62b57e1655e58f126c1df332f38cced7102fed6c5f47dba409ccc924b14R99-R102